### PR TITLE
CircleCI - use PHP 7.1

### PR DESCRIPTION
**Pull request by aidantwoods**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+machine:
+  php:
+    version: 5.6.14
```
Comment (keradus) [inline circle.yml]: i dont find this config to work on osx

Comment (aidantwoods) [inline circle.yml]: Hmm... yeah I think you're right actually – only seeing it mentioning ubuntu now looking at it.

I'll try pull it from homebrew in that case.

File: circle.yml [hunk 2]
```diff
@@ -1,6 +1,11 @@
 machine:
-  php:
-    version: 5.6.14
+  xcode:
+    version: 8.2
```
Comment (keradus) [inline circle.yml]: do we need to configure xcode ?

File: circle.yml [hunk 3]
```diff
@@ -1,6 +1,11 @@
...
+
+dependencies:
+  pre:
+    - brew tap homebrew/homebrew-php
+    - brew install php56
```
Comment (keradus) [inline circle.yml]: could we jump to 7.1 ?

File: circle.yml (hunk 2 above)
Comment (aidantwoods) [inline circle.yml]: Circle docs indicate xcode is bundled at `7.x`, latest homebrew requires `>=8`

File: circle.yml (hunk 3 above)
Comment (aidantwoods) [inline circle.yml]: Yup, you're happy for no tests on `5.x` for Mac then?

Comment (keradus) [inline circle.yml]: yep, as long as we could have only one compiler version let it be newest for biggest code execution

**Comment by aidantwoods** [comment]
Circle UI is very overwhelming ;p can you spot where that `brew install` is? `phpunit` reads the php version as `5.5` :/

**Comment by aidantwoods** [comment]
Two remaining failures are expected until #2933

File: circle.yml [hunk 4]
```diff
@@ -1,11 +1,18 @@
+machine:
+  xcode:
```
Comment (keradus) [inline circle.yml]: all the other has 4 spaces indent

**Comment by keradus** [comment]
Thank you @aidantwoods

**Pull request merged**